### PR TITLE
Replaces direct text output with to_chat

### PR DESCRIPTION
**Pull request by ghost**
OH GOD

port of https://github.com/tgstation/tgstation/pull/24922

Regex is ```(\b(?!\blog|diary|bit\b)\w+\b) << ((?!browse|'|hotkey|adminhotkeys|link|other|output|sound).+?)(?:\/\/.*)?$```.

It has trouble with multilines, you WILL need to fix them manually.

### CODE REVIEW

File: code/datums/martial.dm [hunk 1]
```diff
@@ -457,7 +457,7 @@
 /obj/item/weapon/sleeping_carp_scroll/attack_self(mob/living/carbon/human/user)
 	if(!istype(user) || !user)
 		return
-	user << "<span class='sciradio'>You have learned the ancient martial art of the Sleeping Carp! Your hand-to-hand combat has become much more effective, and you are now able to deflect any projectiles \
+	to_chat(user, "<span class='sciradio'>You have learned the ancient martial art of the Sleeping Carp! Your hand-to-hand combat has become much more effective, and you are now able to deflect any projectiles \)
```
Comment (Amelia0010) [inline code/datums/martial.dm]: \</span>

File: code/game/gamemodes/antag_spawner.dm [hunk 2]
```diff
@@ -296,11 +296,11 @@
 	new_objective2.explanation_text = "[objective_verb] everyone else \
 		while you're at it."
 	S.mind.objectives += new_objective2
-	S << S.playstyle_string
-	S << "<B>You are currently not currently in the same plane of \
+	to_chat(S, S.playstyle_string)
+	to_chat(S, "<B>You are currently not currently in the same plane of \)
```
Comment (Amelia0010) [inline code/game/gamemodes/antag_spawner.dm]: \</b>

Edit: also cut off early

File: code/game/gamemodes/changeling/powers/fleshmend.dm [hunk 3]
```diff
@@ -29,9 +29,9 @@
 //Starts healing you every second for 10 seconds.
 //Can be used whilst unconscious.
 /obj/effect/proc_holder/changeling/fleshmend/sting_action(mob/living/user)
-	user << "<span class='notice'>We begin to heal rapidly.</span>"
+	to_chat(user, "<span class='notice'>We begin to heal rapidly.</span>")
 	if(recent_uses > 1)
-		user << "<span class='warning'>Our healing's effectiveness is reduced \
+		to_chat(user, "<span class='warning'>Our healing's effectiveness is reduced \)
```
Comment (Amelia0010) [inline code/game/gamemodes/changeling/powers/fleshmend.dm]: \</span>

edit: also cut off

File: code/game/gamemodes/clock_cult/clock_cult.dm [hunk 4]
```diff
@@ -160,17 +160,17 @@ This file's folder contains:
 			slot = "In your [B.name]"
 	if(slot == "At your feet")
 		new/obj/item/clockwork/slab/starter(get_turf(L))
-	L << "<b>[slot] is a link to the halls of Reebe and your master. You may use it to perform many tasks, but also become oriented with the workings of Ratvar and how to best complete your \
+	to_chat(L, "<b>[slot] is a link to the halls of Reebe and your master. You may use it to perform many tasks, but also become oriented with the workings of Ratvar and how to best complete your \)
```
Comment (Amelia0010) [inline code/game/gamemodes/clock_cult/clock_cult.dm]: \</b>

Edit: also cut off early

**Comment by ShadowDeath6** [comment]
I cba to find the tg pr that did this originally, is this made to improve performance?

**Comment by ghost** [comment]
@ShadowDeath6 preparation for goonchat

**Comment by ghost** [comment]
IT COMPILESSSSS

File: config/admins.txt [hunk 5]
```diff
@@ -1,60 +1,60 @@
...
+# Case is not important for ckey.                                                             #
+# Case IS important for the rank.                                                             #
+# All punctuation (spaces etc) EXCEPT '-', '_' and '@' will be stripped from rank names.      #
+# Ranks can be anything defined in admin_ranks.txt                                            #
+# NOTE: if the rank-name cannot be found in admin_ranks.txt, they will not be adminned! ~Carn #
+# NOTE: syntax was changed to allow hyphenation of ranknames, since spaces are stripped.      #
+###############################################################################################
+Kn0ss0s	=	Head Coder
+Oisin100	=	Head Coder
+Thefastfoodguy	=	Council Member
```
Comment (Amelia0010) [inline config/admins.txt]: 🤔

Comment (ghost) [reply config/admins.txt]: god damn it

**Comment by ghost** [comment]
Tested it locally, everything should work!

File: code/controllers/subsystem/ticker.dm [hunk 6]
```diff
@@ -289,7 +289,7 @@ var/datum/subsystem/ticker/ticker
 					if (!istype(M.loc, /obj/structure/closet/secure_closet/freezer/meat))
 						M.death(0) //no mercy
 					else
-						M << "The freezer wobbles a bit, then stops. You let out a sigh of relief.";
+						to_chat(M, "The freezer wobbles a bit, then stops. You let out a sigh of relief.";)
```
Comment (Amelia0010) [inline code/controllers/subsystem/ticker.dm]: Will this semicolon harm anything?

File: code/game/gamemodes/shadowling/shadowling_abilities.dm [hunk 7]
```diff
@@ -996,29 +996,29 @@
...
 	if(user.incorporeal_move)
 		revert_cast()
-		user << "<span class='warning'>You are not in the same plane of existence. Unphase first.</span>"
+		to_chat(user, "<span class='warning'>You are not in the same plane of existence. Unphase first.</span>")
 		return
 
 	for(var/mob/living/carbon/human/target in targets)
 		if(is_shadow_or_thrall(target))
-			user << "<span class='warning'>You cannot enthrall an ally.<span>"
+			to_chat(user, "<span class='warning'>You cannot enthrall an ally.<span>")
```
Comment (X-TheDark) [inline code/game/gamemodes/shadowling/shadowling_abilities.dm:1004]: `</span>`
But I'm pretty sure this doesn't make any difference whatsoever.

File: code/game/machinery/vending.dm [hunk 8]
```diff
@@ -552,11 +552,11 @@
 
 	if((href_list["vend"]) && (vend_ready))
 		if(panel_open)
-			usr << "<span class='notice'>The vending machine cannot dispense products while its service panel is open!</span>"
+			to_chat(usr, "<span class='notice'>The vending machine cannot dispense products while its service panel is open!</span>")
 			return
 
 		if((!allowed(usr)) && !emagged && scan_id)	//For SECURE VENDING MACHINES YEAH
-			usr << "<span class='warning'>Access denied.</span>"	//Unless emagged of course
+			to_chat(usr, "<span class='warning'>Access denied.</span>"	)
```
Comment (X-TheDark) [inline code/game/machinery/vending.dm]: Mind curing my OCD by untabbing the closing bracket, please?

File: code/modules/admin/admin_investigate.dm [hunk 9]
```diff
@@ -22,7 +22,7 @@
 	var/F = investigate_subject2file(subject)
 	if(!F)
 		return
-	F << "<small>[time_stamp()] \ref[src] ([x],[y],[z])</small> || [src] [message]<br>"
+	to_chat(F, "<small>[time_stamp()] \ref[src] ([x],[y],[z])</small> || [src] [message]<br>")
```
Comment (X-TheDark) [inline code/modules/admin/admin_investigate.dm]: Revert this, admin logs shouldn't be subject to any chat-filters.

File: code/modules/admin/admin_ranks.dm [hunk 10]
```diff
@@ -163,7 +163,7 @@ var/list/admin_ranks = list()								//list of all admin_rank datums
...
...nt-orientated actions. Access to the fun secrets in the secrets panel.\n# +BAN = the ability to ban, jobban and fullban\n# +STEALTH = the ability to stealthmin (make yourself appear with a fake name to everyone but other admins\n# +POSSESS = the ability to possess objects\n# +BUILD (or +BUILDMODE) = the ability to use buildmode\n# +SERVER = higher-risk admin verbs and abilities, such as those which affect the server configuration.\n# +DEBUG = debug tools used for diagnosing and fixing problems. It's useful to give this to coders so they can investigate problems on a live server.\n# +VAREDIT = everyone may view viewvars/debugvars/whatever you call it. This keyword allows you to actually EDIT those variables.\n# +RIGHTS (or +PERMISSIONS) = allows you to promote and/or demote people.\n# +SOUND (or +SOUNDS) = allows you to upload and play sounds\n# +SPAWN (or +CREATE) = mob transformations, spawning of most atoms including mobs (high-risk atoms, e.g. blackholes, will require the +FUN flag too)\n# +EVERYTHING (or +HOST or +ALL) = Simply gives you everything without having to type every flag\n\n# DO NOT EDIT THIS FILE DIRECTLY\n# IT IS AUTOMATICALLY GENERATED FROM THE SERVER DATABASE\n")
```
Comment (X-TheDark) [inline code/modules/admin/admin_ranks.dm]: Revert. File writing shouldn't be subject to chatfilters.

File: code/modules/admin/admin_ranks.dm [hunk 11]
```diff
@@ -173,14 +173,14 @@ var/list/admin_ranks = list()								//list of all admin_rank datums
 				var/rank_group = text2num(query_ranks.item[3])
 				if(lastGroup != rank_group)
 					lastGroup = rank_group
-					ranksFile << " "
-				ranksFile << "[rank_name]\t=\t[rank_byond]"
+					to_chat(ranksFile, " ")
+				to_chat(ranksFile, "[rank_name]\t=\t[rank_byond]")
```
Comment (X-TheDark) [inline code/modules/admin/admin_ranks.dm]: Revert, as per above.

File: code/modules/admin/admin_ranks.dm [hunk 12]
```diff
@@ -173,14 +173,14 @@ var/list/admin_ranks = list()								//list of all admin_rank datums
...
+			to_chat(adminsFile, "###############################################################################################\n# Basically, ckey goes first. Rank goes after the '='                                         #\n# Case is not important for ckey.                                                             #\n# Case IS important for the rank.                                                             #\n# All punctuation (spaces etc) EXCEPT '-', '_' and '@' will be stripped from rank names.      #\n# Ranks can be anything defined in admin_ranks.txt                                            #\n# NOTE: if the rank-name cannot be found in admin_ranks.txt, they will not be adminned! ~Carn #\n# NOTE: syntax was changed to allow hyphenation of ranknames, since spaces are stripped.      #\n###############################################################################################\n\n# DO NOT EDIT THIS FILE DIRECTLY\n# IT IS AUTOMATICALLY GENERATED FROM THE SERVER DATABASE\n")
```
Comment (X-TheDark) [inline code/modules/admin/admin_ranks.dm]: Revert.

File: code/modules/admin/admin_ranks.dm [hunk 13]
```diff
@@ -190,8 +190,8 @@ var/list/admin_ranks = list()								//list of all admin_rank datums
 				var/rank_group = text2num(query_admin.item[3])
 				if(lastGroup != rank_group)
 					lastGroup = rank_group
-					adminsFile << " "
-				adminsFile << "[name]\t=\t[rank]"
+					to_chat(adminsFile, " ")
+				to_chat(adminsFile, "[name]\t=\t[rank]")
```
Comment (X-TheDark) [inline code/modules/admin/admin_ranks.dm]: Revert.

File: code/modules/admin/verbs/BrokenInhands.dm [hunk 14]
```diff
@@ -28,7 +28,7 @@
 	if(text)
 		var/F = file("broken_icons.txt")
 		fdel(F)
-		F << text
-		world << "Completely successfully and written to [F]"
+		to_chat(F, text)
```
Comment (X-TheDark) [inline code/modules/admin/verbs/BrokenInhands.dm]: File operation, no chatfilters.

File: code/modules/awaymissions/maploader/swapmaps.dm [hunk 15]
```diff
@@ -383,7 +383,7 @@ swapmap
 	proc/Save()
 		if(id==src) return 0
 		var/savefile/S=mode?(new):new("map_[id].sav")
-		S << src
+		to_chat(S, src)
```
Comment (X-TheDark) [inline code/modules/awaymissions/maploader/swapmaps.dm]: Savefile operation, remove chatfilter.

File: code/modules/awaymissions/maploader/writer.dm [hunk 16]
```diff
@@ -36,7 +36,7 @@ dmm_suite{
 			fdel("[map_name].dmm")
 			}
 		var/saved_map = file("[map_name].dmm")
-		saved_map << file_text
+		to_chat(saved_map, file_text)
```
Comment (X-TheDark) [inline code/modules/awaymissions/maploader/writer.dm]: File operation, remove chatfilter.

File: code/modules/client/client_procs.dm [hunk 17]
```diff
@@ -79,16 +79,16 @@
...
 		return
 
 	if(prefs.afreeze && !holder)
-		src << "<span class='userdanger'>You are frozen by an administrator.</span>"
+		to_chat(src, "<span class='userdanger'>You are frozen by an administrator.</span>")
 		return
 	//Logs all hrefs
 	if(config && config.log_hrefs && href_logfile)
-		href_logfile << "<small>[time2text(world.timeofday,"hh:mm")] [src] (usr:[usr])</small> || [hsrc ? "[hsrc] " : ""][href]<br>"
+		to_chat(href_logfile, "<small>[time2text(world.timeofday,"hh:mm")] [src] (usr:[usr])</small> || [hsrc ? "[hsrc] " : ""][href]<br>")
```
Comment (X-TheDark) [inline code/modules/client/client_procs.dm]: Logging. Remove chatfilter.

File: code/world.dm [hunk 18]
```diff
@@ -30,7 +30,7 @@ var/global/list/map_transition_config = MAP_TRANSITION_CONFIG
 	diaryofmeanpeople = file("data/logs/[date_string] Attack.log")
 	diary << "\n\nStarting up. [time2text(world.timeofday, "hh:mm.ss")]\n---------------------"
 	diaryofmeanpeople << "\n\nStarting up. [time2text(world.timeofday, "hh:mm.ss")]\n---------------------"
-	admindiary << "\n\nStarting up. [time2text(world.timeofday, "hh:mm.ss")]\n---------------------"
+	to_chat(admindiary, "\n\nStarting up. [time2text(world.timeofday, "hh:mm.ss")]\n---------------------")
```
Comment (X-TheDark) [inline code/world.dm]: File operation, revert.

File: code/world.dm [hunk 19]
```diff
@@ -355,7 +355,7 @@ var/inerror = 0
 /world/proc/save_mode(the_mode)
 	var/F = file("data/mode.txt")
 	fdel(F)
-	F << the_mode
+	to_chat(F, the_mode)
```
Comment (X-TheDark) [inline code/world.dm]: File operation, revert.

File: code/world.dm [hunk 20]
```diff
@@ -558,7 +558,7 @@ var/rebootingpendingmapchange = 0
 	mapchanging = 1
 	log_game("Changing map to [VM.name]([VM.friendlyname])")
 	var/file = file("setnewmap.bat")
-	file << "\nset MAPROTATE=[VM.name]\n"
+	to_chat(file, "\nset MAPROTATE=[VM.name]\n")
```
Comment (X-TheDark) [inline code/world.dm]: File operation, revert.

**Review by X-TheDark — CHANGES_REQUESTED** [review]
Logging functions should be exempt from using any sort of chatfilter.

File: code/game/gamemodes/shadowling/shadowling_abilities.dm (hunk 7 above)
Comment (ghost) [reply code/game/gamemodes/shadowling/shadowling_abilities.dm:1004]: it doesn't

File: code/modules/admin/admin_ranks.dm (hunk 11 above)
Comment (ghost) [reply code/modules/admin/admin_ranks.dm]: i reverted the entire file

**Review by X-TheDark — APPROVED** [review]

**Comment by X-TheDark** [comment]
Going to merge this because it touches a shitton of files.

**Pull request merged**